src/hooks/events.ts

```diff
@@ -25,13 +29,24 @@ export interface VenueEventsData {
 export const useVenueEvents: ReactHook<VenueEventsProps, VenueEventsData> = ({
```

Comment:
`useVenueEvents` is called in many places across the platform, and the events it fetches/returns depend on the `venueId`s provided. As is currently implemented here, that functionality isn't included/handled at all.

Reply:
So someone should check if filtering the events by venueId would still at the end would provide the speedup overall I guess.

Note that this PR is just an attempt to help to make sparkle usable for the ongoing event. It is provided as a prototype for you, sparkle folks, to try/improve etc, to save the event. If you see something to improve, just join/push.

Reply:
also might be worth adding a condition and use cached version only if many venueIds are requested, and if not -- resort to non-cached version

Reply:
> Note that this PR is just an attempt to help to make sparkle usable for the ongoing event. It is provided as a prototype for you, sparkle folks, to try/improve etc, to save the event. If you see something to improve, just join/push.

We're working on our own more specific/better performance fixes. I spent the time to review this PR out of good will and to see if it would help. In its current state, it's not really usable. It's not worth me diverting my efforts from proper bigger better fixes to try and piece together that half implementation provided here.

Reply:
@0xdevalias Thank you for reviewing this PR. Until better fixes are available, would it be possible to merge this now and later replace it with something better?

Reply:
@soichih wouldn't it need at least addition of filtering for the `venueIds` given to the call, and thus also just returning if none was requested?

Reply:
Yes, that's true. I should filter them by requested venueIds. I will update